src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -4230,7 +4353,8 @@ bool QgsSpatiaLiteProvider::addFeatures( QgsFeatureList &flist, Flags flags )
           {
             unsigned char *wkb = nullptr;
             int wkb_size;
-            QByteArray featureWkb = feature->geometry().asWkb();
+            const QgsGeometry convertedGeom( convertToProviderType( feature->geometry() ) );
```

Comment:
shouldn't that be called too in QgsSpatiaLiteProvider::changeGeometryValues() ?

Reply:
yes, I'll make the changes.